### PR TITLE
add pop up modal form

### PR DESCRIPTION
**Pull request by swanie21**
Add form modal on the toggle "Add Spike" button

Fix the bug with adding spikes - calling two functions within an input field was breaking the adding spike functionality

### CODE REVIEW

File: lib/components/Spike.js [hunk 1]
```diff
@@ -5,42 +5,53 @@ import moment from 'moment';
...
-          />
-          <input
-            className="SubmitButton"
-            type='submit'
-            onClick={(e) => {
-              e.preventDefault();
-              toggleForm(e)
+      <div className='modal'>
+        <h1 className='FormTitle'>Submit Spike</h1>
+        <div id='SpikeForm'>
```
Comment (kccrs) [inline lib/components/Spike.js]: Do we need a div for the form?  I think the form creates its own container.  We seem to have a lot of divs here.

File: lib/components/Spike.js [hunk 2]
```diff
@@ -5,42 +5,53 @@ import moment from 'moment';
...
+          >
+            <input
+              placeholder='Topic Title'
+              name='title'
+            />
+            <textarea
+              placeholder='Description'
+              name='description'
+            />
+            <div className="ButtonContainer">
```
Comment (kccrs) [inline lib/components/Spike.js]: Maybe we can make this a "section" to use more semantic elements and reduce our use of divs?

File: lib/components/Spike.js [hunk 3]
```diff
@@ -5,42 +5,53 @@ import moment from 'moment';
...
-          <input
-            className="SubmitButton"
-            type='submit'
-            onClick={(e) => {
-              e.preventDefault();
-              toggleForm(e)
+      <div className='modal'>
+        <h1 className='FormTitle'>Submit Spike</h1>
+        <div id='SpikeForm'>
+          <form
```
Comment (kccrs) [inline lib/components/Spike.js]: Do we need a div for the form? I think the form creates its own container. We seem to have a lot of divs here.

Comment (kerrd89) [inline lib/components/Spike.js]: I think we just have the extra opening div with the ID of SpikeForm, i will update it and push back up.  Good catch.

File: lib/components/Spike.js (hunk 2 above)
Comment (kerrd89) [inline lib/components/Spike.js]: we can change this at a later point, but this shouldn't be a one-off thing we do just in this file.  If we decide to use sections, we should decide when we want to use sections and implement them everywhere.

**Pull request merged**